src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp

```diff
@@ -40,18 +40,69 @@ void check_level_zero_attributes_match(const IONodeDescriptor& nodeDescriptor,
             "Given: " + std::to_string(ovDimensions.size()));
     }
 
-    for (size_t index = 0; index < ovDimensions.size(); ++index) {
-        if (ovDimensions[index] != zeDescriptor.info.dims[index] && !nodeDescriptor.originalShape.is_dynamic()) {
-            OPENVINO_THROW("Shape mismatch for parameter " + name);
-        }
-    }
     for (size_t index = ovDimensions.size(); index < ZE_MAX_GRAPH_ARGUMENT_DIMENSIONS_SIZE; ++index) {
         if (zeDescriptor.info.dims[index] != 0 && zeDescriptor.info.dims[index] != 1) {
             OPENVINO_THROW("Shape mismatch for parameter " + name);
         }
     }
 }
 
+size_t get_batch_size_for_node(const IONodeDescriptor& nodeDescriptor,
+                               const ZeroExecutor::ArgumentDescriptor& zeDescriptor) {
+    const std::vector<size_t>& ovDimensions = nodeDescriptor.originalShape.get_shape();
+    switch (zeDescriptor.info.deviceLayout) {
+    case ZE_GRAPH_ARGUMENT_LAYOUT_NCHW:
+    case ZE_GRAPH_ARGUMENT_LAYOUT_NHWC:
+    case ZE_GRAPH_ARGUMENT_LAYOUT_NCDHW:
+    case ZE_GRAPH_ARGUMENT_LAYOUT_NDHWC:
+    case ZE_GRAPH_ARGUMENT_LAYOUT_NC:
+        if ((ovDimensions[0] == zeDescriptor.info.dims[0]) && (ovDimensions[0] != 1)) {
+            OPENVINO_THROW("Batching on the plugin is not used, batching is handled by the compiler");
+        } else {
+            return ovDimensions[0];
+        }
+        break;
+    default:
+        OPENVINO_THROW("Batching on the plugin is working only when batching is found on 0th dimension");
+    }
+
+    return 1;
+}
+
+std::optional<size_t> get_batch_size(
+    const NetworkMetadata& metadata,
+    const std::unordered_map<std::string, ZeroExecutor::ArgumentDescriptor>& executorInputDescriptors,
+    const std::unordered_map<std::string, ZeroExecutor::ArgumentDescriptor>& executorOutputDescriptors) {
+    std::set<size_t> batch_size;
+
+    for (const std::string& inputName : metadata.inputNames) {
+        if (!executorInputDescriptors.count(inputName)) {
+            OPENVINO_THROW("Invalid graph input descriptor key: " + inputName);
+        }
+        batch_size.insert(
+            get_batch_size_for_node(metadata.parameters.at(inputName), executorInputDescriptors.at(inputName)));
+    }
+
+    for (const std::string& outputName : metadata.outputNames) {
+        if (!executorOutputDescriptors.count(outputName)) {
+            OPENVINO_THROW("Invalid graph output descriptor key: " + outputName);
+        }
+        batch_size.insert(
+            get_batch_size_for_node(metadata.results.at(outputName), executorOutputDescriptors.at(outputName)));
+    }
+
+    if (batch_size.size() != 1) {
+        OPENVINO_THROW("Batching on the plugin is working only when have same value for all tensors!");
```

Comment:
```suggestion
        OPENVINO_THROW("Batching works only when we have the same batch size for all tensors!");
```
This observation is not limited to our implementation of batching inside the plugin - batching as a concept doesn't make sense if there is a mismatch in the batch size.

Reply:
done, thanks!